net/http/isvalidcookievalue.c

```diff
@@ -0,0 +1,39 @@
+/*-*- mode:c;indent-tabs-mode:nil;c-basic-offset:2;tab-width:8;coding:utf-8 -*-│
+│vi: set net ft=c ts=2 sts=2 sw=2 fenc=utf-8                                :vi│
+╞══════════════════════════════════════════════════════════════════════════════╡
+│ Copyright 2021 Justine Alexandra Roberts Tunney                              │
+│                                                                              │
+│ Permission to use, copy, modify, and/or distribute this software for         │
+│ any purpose with or without fee is hereby granted, provided that the         │
+│ above copyright notice and this permission notice appear in all copies.      │
+│                                                                              │
+│ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL                │
+│ WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED                │
+│ WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE             │
+│ AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL         │
+│ DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR        │
+│ PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER               │
+│ TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR             │
+│ PERFORMANCE OF THIS SOFTWARE.                                                │
+╚─────────────────────────────────────────────────────────────────────────────*/
+#include "libc/str/str.h"
+#include "net/http/http.h"
+
+/**
+ * Returns true if string is a valid cookie value
+ * (any ASCII excluding control char, whitespace,
+ * double quotes, comma, semicolon, and backslash).
+ *
+ * @param n if -1 implies strlen
+ */
+bool IsValidCookieValue(const char *s, size_t n) {
+  size_t i;
+  if (!n) return false;
+  if (n == -1) n = s ? strlen(s) : 0;
+  for (i = 0; i < n; ++i) {
```

Comment:
What about `DQUOTE *cookie-octet DQUOTE`? https://datatracker.ietf.org/doc/html/rfc6265#section-4.1.1

Reply:
Yes, it's allowed, but only needs to be checked on the receiving side, as the method itself will generate non-quoted `*cookie-octet` (unless I misunderstood what you were suggesting).

Reply:
Ah I see.